### DISCUSSION TITLE
date.setZone is not a function in Edge

### DISCUSSION THREAD

**Discussion by galaczi** [adonisjs/core#3564] [Help] [answered]
I have the following input field: 

```
<input type="datetime-local" name="{{name}}" value="{{ inputDate.setZone('UTC+1').toFormat("yyyy-MM-dd'T'hh:mm") ?? '' }}">
```

It is displayed correctly in my edge template. However upon form submission I get the error ``inputDate.setZone is not a function``. 

I am confused why is it even checked when I submit the form. A hint might be that I have ``app/Middleware/SilentAuth.ts:19:5`` in my stack trace?

**Comment by RomainLanz** [comment, marked as answer]
Hey @galaczi! 👋🏻 

Could you please share your controller code?
It seems that you are displaying this view again, and `inputDate` is not set.

**Reply by galaczi** [reply]
Looks like the form validation failed and I reassigned the inputDate variable with the value from flashMessages, which was just a string. Hence setZone does not exist on it.
